### ISSUE TITLE
extraneous header tags when using horizontal rule

### ISSUE THREAD

**Issue by kentor** [npm/marky-markdown#389]
I think this is a bug:

markdown source:
```
<p></p>

---

<p></p>
```

Preview:

<p></p>

---

<p></p>

Github produces this html (as expected):
```html
<p></p>
<hr>
<p></p>
```

marky-markdown (10.1.4):
```html
<p></p>
<h2></h2>
<hr>
<p></p>
```

Notice the extra `<h2>` tags.

Gets weirder if I have more stuff:

```markdown
<p>
  <a href="">asdfas</a>
</p>

---

<p></p>
```

produces

```html
<h2></h2>
<p>
  <a href="">asdfas</a>
</p>
<h2></h2>
<hr>
<p></p>
```

**Comment by revin** [comment]
Hey @kentor yep, you're right, this is indeed a bug. Thanks! I'll take a look.

**Comment by revin** [comment]
OK, this is fixed in 11.0.0, just released. You can verify it on https://revin.github.io/marky-markdown/ if you like, while we wait for it to be deployed to the npm site.

**Comment by kentor** [comment]
Thanks for the quick fix. Really appreciate it.
